### PR TITLE
Rewrite firmware version checks

### PR DESCRIPTION
**Pull request by 13r0ck**
1. The GUI notification that firmware needs to be updated is now part of the button, not the loading spinner, this removes the need for the board-not-updated signal
2. is_testing_mode no longer needs to be passed stored by the backend. It is sent as part of the refresh loop.
3. is_launch_updated is now checked when the `board` is created rather than every refresh loop. This is a lot more efficient.
4. When in testing mode and the keyboard is not updated, the board is now added to the application, this improves a bug that production was seeing where plugging in a keyboard, then replugging in bootloader mode would not detect the bootloader keyboard.

### CODE REVIEW

File: widgets/src/keyboard_backlight_widget.rs
Comment (13r0ck) [reply widgets/src/keyboard_backlight_widget.rs]: ok [4fa4070](https://github.com/pop-os/keyboard-configurator/pull/175/commits/4fa4070860574dafb7abc4b12d8970a5d312cd53) :)

**Comment by leviport** [comment]
If I have a Launch that needs an update plugged into an open firmware laptop (oryp9 or serw13 in this case) and then start the configurator in `--launch-test` mode, it tells me that my laptop needs a keyboard firmware update:


If I then unplug and re-plug the Launch, the button is restored to normal:


This is pretty minor and non-blocking since we use Meerkats for the testing stations, but I wanted to bring it up since I found it. I'm still investigating some other weird behavior that I don't quite understand yet, but I'll approve if this is the only issue with this PR.

**Comment by leviport** [comment]
I also noticed that the prompt says to flash "launch_1.0" even if I'm flashing a launch_1.3. Since they use the same firmware, it might make more sense to say "flash launch_1".

**Comment by leviport** [comment]
I figured out what was giving me so much trouble. This doesn't work if `avrdude` is not installed. I don't think it makes sense to add `avrdude` as a dependency, since the vast majority of users won't use this functionality, but I do think it could save Production a lot of potential debugging time if there was a check and a warning if `avrdude` is not installed.

**Comment by leviport** [comment]
A deb package for testing mode is an interesting idea. That would also make it easier to set up testing machines, since we wouldn't have to make custom .desktop files. Is this something you want to take care of before I approve this? I could also file an issue and we could come back to it later.

**Review by leviport — DISMISSED** [review]
:metal:

**Review by ids1024 — CHANGES_REQUESTED** [review]
A `.postinst` hook that changes a file belonging to another package isn't the right way to do this.

Maybe the package could provide a second desktop file that also has a distinct name.

**Comment by 13r0ck** [comment]
fixed [1449377](https://github.com/pop-os/keyboard-configurator/pull/175/commits/14493774851f97587ae5f706fe7770292819951f)

**Comment by 13r0ck** [comment]
[1449377](https://github.com/pop-os/keyboard-configurator/pull/175/commits/14493774851f97587ae5f706fe7770292819951f)

File: Makefile
Comment (ids1024) [inline Makefile:49]: Installing this unconditionally in the makefile may cause confusion for users of the AUR package or any other packaging outside of Pop.

You should be able to just use something like this in the `.install` file, without installing it here:

```
linux/com.system76.keyboardconfiguratorlaunchtest.desktop usr/share/applications
```

Comment (13r0ck) [reply Makefile]: [09246d4](https://github.com/pop-os/keyboard-configurator/pull/175/commits/09246d402b64b2584b2c1baeb5292b42f7cd7751)

File: debian/test-launch-system76-keyboard-configurator.install
Comment (ids1024) [inline debian/test-launch-system76-keyboard-configurator.install:1]: The part on the right here is installation directory, so this will result in a `com.system76.keyboardconfiguratorlaunchtest.desktop` directory containing a file of the same name.

(Yes, Debian packaging is an arcane dark magic.)

Comment (13r0ck) [reply debian/test-launch-system76-keyboard-configurator.install]: Ah I got a little too tab complete happy. Fixed. Thanks

**Review by ids1024 — APPROVED** [review]

File: linux/com.system76.keyboardconfiguratorlaunchtest.desktop
Comment (leviport) [inline linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: Screenshot from 2023-04-20 15-13-28

Can we give this one a different name? Otherwise they're a pain to figure out.

**Review by leviport — CHANGES_REQUESTED** [review]

File: linux/com.system76.keyboardconfiguratorlaunchtest.desktop
Comment (13r0ck) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: I actually have no idea why it is doing that

Comment (13r0ck) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: On cosmic they are different

File: debian/test-launch-system76-keyboard-configurator.install
Comment (ids1024) [reply debian/test-launch-system76-keyboard-configurator.install]: I don't know how many times I've gotten `.install` files wrong the first time I tried to write them. :laughing:

File: linux/com.system76.keyboardconfiguratorlaunchtest.desktop
Comment (leviport) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: They're also the same in app library.

Comment (ids1024) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: Weird. Not sure how that would happen unless its makes assumptions that there aren't multiple desktop files that `Exec` the same binary (even with different arguments). Which wouldn't make much sense. Or some kind of caching (but what is cached?).

The two `.desktop` files have the right `Name=` and `Exec=` lines as installed in `/usr/share/applications`?

Comment (leviport) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: They have different Exec lines, but identical Name lines
Screenshot from 2023-04-20 15-53-17

Comment (leviport) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: Oh wait, I was on an old commit. Updating now, and sorry if this ends up being the reason :sweat_smile:

Comment (13r0ck) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: I just did a clean install of pop, ad installed it, they show as two different names. Something is weird with your machine @leviport

Comment (leviport) [reply linux/com.system76.keyboardconfiguratorlaunchtest.desktop:2]: Yep, all good now. Sorry for the false alarm. 
Screenshot from 2023-04-20 15-56-24
Screenshot from 2023-04-20 15-56-38

**Review by leviport — APPROVED** [review]

**Pull request merged**
